das/worker.go

```diff
@@ -54,6 +56,7 @@ func (w *worker) run(
 			break
 		}
 		w.setResult(curr, err)
+		metrics.observeSample(ctx, h, time.Since(start), err)
```

Comment:
Hmmm... this looks a lot like synchronous metrics

Reply:
It is.